std/format/write.d

```diff
@@ -79,129 +85,219 @@ module std.format.write;
     import std.format.spec : singleSpec;
 
-    auto w = appender!string();
-    auto spec = singleSpec("%d");
-    formatValue(w, 1337, spec);
+    auto w1 = appender!string();
+    auto spec1 = singleSpec("%d");
+    formatValue(w1, -1337, spec1);
+
+    assert(w1.data == "-1337");
 
-    assert(w.data == "1337");
+    auto w2 = appender!string();
+    auto spec2 = singleSpec("%x");
+    formatValue(w2, -1337, spec2);
+
+    assert(w2.data == "fffffac7");
 }
 
-/// Floating-point values are formatted like $(REF printf, core, stdc, stdio)
+/**
+Floating-point values are formatted in natural notation with `%f`, in
+scientific notation with `%e`, in short notation with `%g`, and in
+hexadecimal scientific notation with `%a`. They are rounded according
+to the current rounding mode of the floating point unit, if available.
+ */
 @safe unittest
 {
     import std.array : appender;
     import std.format.spec : singleSpec;
 
-    auto w = appender!string();
-    auto spec = singleSpec("%.1f");
-    formatValue(w, 1337.7, spec);
+    auto w1 = appender!string();
+    auto spec1 = singleSpec("%.3f");
+    formatValue(w1, 1337.7779, spec1);
+
+    assert(w1.data == "1337.778");
+
+    auto w2 = appender!string();
+    auto spec2 = singleSpec("%.3e");
+    formatValue(w2, 1337.7779, spec2);
+
+    assert(w2.data == "1.338e+03");
 
-    assert(w.data == "1337.7");
+    auto w3 = appender!string();
+    auto spec3 = singleSpec("%.3g");
+    formatValue(w3, 1337.7779, spec3);
+
+    assert(w3.data == "1.34e+03");
+
+    auto w4 = appender!string();
+    auto spec4 = singleSpec("%.3a");
+    formatValue(w4, 1337.7779, spec4);
+
+    assert(w4.data == "0x1.4e7p+10");
 }
 
 /**
- * Individual characters (`char, `wchar`, or `dchar`) are formatted as
- * Unicode characters with `%s` and as integers with integral-specific format
- * specs.
+Individual characters (`char`, `wchar`, or `dchar`) are formatted as
+Unicode characters with `%s` and `%c` and as integers (`ubyte`,
+`ushort`, `uint`) with all other format characters. With
+$(MREF_ALTTEXT compound specifiers, std,format) characters are
+treated special.
  */
 @safe pure unittest
 {
     import std.array : appender;
     import std.format.spec : singleSpec;
 
-    auto w = appender!string();
-    auto spec = singleSpec("%c");
-    formatValue(w, 'a', spec);
+    auto w1 = appender!string();
+    auto spec1 = singleSpec("%c");
+    formatValue(w1, 'ì', spec1);
 
-    assert(w.data == "a");
+    assert(w1.data == "ì");
+
+    auto w2 = appender!string();
+    auto spec2 = singleSpec("%#x");
+    formatValue(w2, 'ì', spec2);
+
+    assert(w2.data == "0xec");
 }
 
-/// Strings are formatted like $(REF printf, core, stdc, stdio)
+/**
+Strings are formatted as a sequence of characters with `%s`.
+Non-printable characters are not escaped. With a compound specifier
+the string is treated like a range of characters. With $(MREF_ALTTEXT
+compound specifiers, std,format) strings are treated special.
```

Comment:
```suggestion
compound specifiers, std,format) strings are treated differently.
```

Reply:
Also, why is this last sentence necessary?

Reply:
That's this annoying "feature", that strings and characters are treated like literals when they appear inside `%(...%)`.